avalon/nuke/lib.py

```diff
@@ -3,82 +3,283 @@
 import nuke
 import re
 import logging
-import toml
+import six
+from ..vendor import clique
 
 log = logging.getLogger(__name__)
 
 
 @contextlib.contextmanager
 def maintained_selection():
+    """Maintain selection during context
+
+    Example:
+        >>> with maintained_selection():
+        ...     node['selected'].setValue(True)
+        >>> print(node['selected'].value())
+        False
+    """
+    nodes = nuke.allNodes()
     previous_selection = nuke.selectedNodes()
     try:
         yield
     finally:
         # unselect all selection in case there is some
-        current_seletion = nuke.selectedNodes()
-        [n['selected'].setValue(False) for n in current_seletion]
+        [n['selected'].setValue(False) for n in nodes]
         # and select all previously selected nodes
         if previous_selection:
-            [n['selected'].setValue(True) for n in previous_selection]
+            [n['selected'].setValue(True)
+             for n in nodes
+             if n.name() in previous_selection]
+
+
+def reset_selection():
+    """Deselect all selected nodes
+    """
+    nodes = nuke.allNodes()
+    [n['selected'].setValue(False) for n in nodes]
 
 
-def add_avalon_tab_knob(node):
-    """Adding a tab and a knob into a node
+def select_nodes(nodes):
+    """Selects all inputed nodes
+
+    Arguments:
+        nodes (list): nuke nodes to be selected
     """
+    assert isinstance(nodes, (list, tuple)), "nodes has to be list or tuple"
+
+    [n['selected'].setValue(True) for n in nodes]
```

Comment:
This is better written as:
```python
for node in nodes:
    node['selected'].setValue(True)
```

A list comprehension should not be used to perform actions or have those side effects, see [here](https://docs.python-guide.org/writing/style/#short-ways-to-manipulate-lists):

> Never use a list comprehension just for its side effects.